include/votca/tools/spline.h

```diff
@@ -152,41 +149,41 @@ class Spline
      * \brief Get the grid array x
      * \return pointer to the corresponding array
      */
-    ub::vector<double> &getX() {return _r; }
+    Eigen::VectorXd &getX() {return _r; }
 
     /**
      * \brief Get the spline data _f
      * \return pointer to the corresponding array
      */
-    ub::vector<double> &getSplineF() { return _f; }
+    Eigen::VectorXd &getSplineF() { return _f; }
 
     /**
      * \brief Get second derivatives (cubic splines)
      * \return pointer to the corresponding array
      */
-    ub::vector<double> &getSplineF2() { return _f2; }
+    Eigen::VectorXd &getSplineF2() { return _f2; }
     
 protected:
     eBoundary _boundaries;
     // the grid points
-    ub::vector<double> _r;
+    Eigen::VectorXd _r;
     // y values of grid points
-    ub::vector<double> _f;
+    Eigen::VectorXd _f;
     // second derivatives of grid points
-    ub::vector<double> _f2;
+    Eigen::VectorXd _f2;
 };
 
 template<typename vector_type1, typename vector_type2>
 inline void Spline::Calculate(vector_type1 &x, vector_type2 &y)
 {
-    for(size_t i=0; i<x.size(); ++i)
+    for(int i=0; i<x.size(); ++i)
         y(i) = Calculate(x(i));
```

Comment:
Again I think you will run into problems with this if you are comparing an int with a size_t    you may need to cast the int to a size_t before comparing